zulipterminal/model.py

```diff
@@ -516,6 +516,59 @@ def send_stream_message(self, stream: str, topic: str, content: str) -> bool:
             notify_if_message_sent_outside_narrow(composition, self.controller)
         return message_was_sent
 
+    def is_unauthorised_to_post_in_stream(self, stream_id: int) -> Optional[str]:
+        """
+        Identify whether a user is authorised to send message to the given
+        stream or not using stream_post_policy and/or is_announcement_policy
+        and accordingly update the footer
+        """
+        stream_dict = self.stream_dict[stream_id]
+        user_info = self._all_users_by_id.get(self.user_id, None)
+        if not user_info:
+            # Check whether the given user id is valid or not
+            return "User does not exist"
+        if (
+            stream_dict.get("is_announcement_only")
+            or stream_dict.get("stream_post_policy") == 2
+        ):
+            # For admins and owners only (not using user role)
+            if (
+                user_info.get("is_admin")
+                or user_info.get("is_owner")
+                and user_info.get("is_guest") is False
+            ):
+                return None
+            else:
+                return "Only Admins and owners can type"
+        elif stream_dict.get("stream_post_policy") is not None:
+            # is_announcement_only is deprecated in Zulip 3.0+ -> stream_post_policy (ZFL 1)
+            if stream_dict.get("stream_post_policy") == 4:
+                # Check for moderators only stream
+                if (
+                    user_info.get("role", 400) <= 300
+                    or user_info.get("is_admin") is True
+                    or user_info.get("is_owner") is True
+                ) and (user_info.get("is_guest") is False):
```

Comment:
This may be clearer if we hoist the `.get` results into variables above the conditionals.

Reply:
Alright, I'll make the required changes.